### PR TITLE
Tutorial on Feature Ablation

### PR DESCRIPTION
**Pull request by bilalsal**
As requested in T58506541 a tutorial utilizing FeatureAblation in Captum to interpret a computer vision model from torchvision.
The resnet18 is selected as a model (in line with the IG tutorial).
A sample image from PASCAL VOC is selected because of the availability of segmentation masks to be used as feature ablation masks.

For test plan and example attribution results refer to internal diff D19152391

### CODE REVIEW

**Comment by orionr** [comment]
@bilalsal, would you mind entering the task details in the description for this? Much appreciated.

**Comment by orionr** [comment]
Actually putting what you have in the internal diff out here would be amazing.

**Comment by orionr** [comment]
@NarineK and @vivekmig can you review so @bilalsal can land today? Much appreciated!

File: tutorials/Resnet_TorchVision_Ablation.ipynb [hunk 1]
```diff
@@ -0,0 +1,768 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Feature Ablation to Inspect Resnet Decisions"
```
Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: `Resnet` -> `ResNet`

File: tutorials/Resnet_TorchVision_Ablation.ipynb [hunk 2]
```diff
@@ -0,0 +1,768 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook tutorials demonstrates how feature ablation in Captum can be applied to inspect computer vision models. \n",
```
Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: `notebook tutorials` -> `notebook tutorial`

File: tutorials/Resnet_TorchVision_Ablation.ipynb [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+    "\n",
+    "**Task:** Classification into ImageNet-1k categories\n",
+    "\n",
+    "**Model:** Resnet18 trained on ImageNet-1k\n",
+    "\n",
+    "**Data to inspect:** Samples from PASCAL VOC 2012\n",
+    "\n",
+    "**Ablation based on:** Segmentation masks\n",
+    "\n",
+    "We will use the visualization functions in Captum to show how each semantic parts impacts the model output.\n",
```
Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: `semantic parts` -> `semantic part`

File: tutorials/Resnet_TorchVision_Ablation.ipynb [hunk 4]
```diff
@@ -0,0 +1,768 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "root = \"./VOC\"\n",
+    "dataset = torchvision.datasets.VOCSegmentation(root, year='2012', image_set='train', download=False, transform=None, target_transform=None)\n",
```
Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Do you want to set `download=True` here?

File: tutorials/Resnet_TorchVision_Ablation.ipynb [hunk 5]
```diff
@@ -0,0 +1,768 @@
...
...RaXTvH444vMbWuxb2nMs6+6ju/5zjfw7o++n5Mrm/jVIVmn4tHjh5lfOo/+w/ezNFilk6dIkbJ22nDi+DrLnUCIFHMFF1z7Yt743FdyZtjHqy6m8pjyNDrbxZHTKyzu2MNHPvY+Xv3SF7C2dpx77rubue0XYKoRo/GIy5/5fLIsC+4ccWPUmUdD2DTRZaBx/JgBn2fKkjNrp/mLd3yU//TvX8rBA0sUoxJvNV6MgqbPCfCGtGxx4uEVThSn6B4/ymLnch594m4ues1Bbr75Jbz+R/+cN37ny5k77xDLnQG9tYz/8rY/4Df/9Df5jz/7HxkNC2668bksMMcHP/pB7r73Tg4f+RdW+g8w2HQst7sMJwN2zc3z/r/9NBvjPn/zB+/GlJu86XUv57JLd7O0tMyH3/U1nvPad6IXl7hcFqi1nH6ywmo1ptdapNvKUamgdIIysUzKCjEsyVqWjc0RrVab8chhnGAwEkhbkGY5aXTdKKqi2SjeBzBaJ1GLm+VNA0AL2XCsrHNbCJmZD3ZQIaXRCD3tnNalmK1cbCxEMZr3SB3mXpSladwhJt4jY4dSEzI36wMFRPgQNFpCNfjh05acsyWbDY0HrSNPTgo6nU6UBm0d11dzFGd/xqi0QbkD+JhJIjzemFCuKxW01GeVgc65IJGawfSa7e+2ls5NJzlqoIFGr1qXofWhU3+uTcjySuGw7VDGYixZZXB4jPSseYG1VSBK2wpTFdGXMNoaDcL9jL0PJgTnuL4JDy2UOQ2XzXt8JHTWEnWNQylBRUUiC1SaMxyPokleQAh6aU6aW+bmu/RHZ+h151Fpgk4ykmRMx2suOXQpK/0TnDq5wbjMKXyPTpJjzIiNjbtIK41L9tCbd6xXktGZkwwXe5w8eopnXfsMnJcszG/n5PEjVOVpik3DXXffwZ49e7jvrs+jNs/Q6XXplxsknW08vnKYxd4iSnlOrpzilue8mJNnRuhMMyoGKAcbZ1bxIiFthdmMB8+/kKKouPXWT/Pdt7yeMxsbLC70EN0sEg9LJDKWKgTOHuCdaYTaBhdoLzNYRvi7QwSLWgbDMZN+ybXXvIgHH7idXi8BlyGkRCeATQBBV+VsaseFlz6LUQFqVPLa17+IcnSMh7/meOZNL2ZhYcJDXz9Fp1vxku/4Ln7r93+fO+/5BF+4/15cAe/6iz/hVTe/nDe8/nt58
```
Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: `contains three bottles, and two TV monitors,` -> `contains three bottles and two TV monitors,`

Comment (orionr) [reply tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit:  `among he` -> `among the`

Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: `we process its output with` -> `we process the output with`

Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: 
```
`bottle`, and `tvmonitor` and `void` -> `bottle`, `tvmonitor`, and `void`
```

Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: `While they would work, but currently Captum expects consecutive group ids, and would ` -> `While they would work, Captum expects consecutive group ids and would `

Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit:
```
which is  `lotion` -> which is `lotion` (remove a space)
```

Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: `tv monitor`  -> `TV monitor`

Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: `apparently make the monitors`  -> `apparently makes the monitors`

Comment (orionr) [inline tutorials/Resnet_TorchVision_Ablation.ipynb]: Nit: `correspondsing` -> `corresponding`

**Comment by orionr** [comment]
Also if you could rebase on latest master that would be great.

**Comment by NarineK** [comment]
Thank you for the PR @bilalsal ! I've started looking into this and will post comments today!

**Pull request closed without merging**